lib/state/state.js

```diff
@@ -7,17 +9,44 @@ var mage;
 var logger;
 var Archivist;
 
+/**
+ * Default description if none is set on a state object
+ */
 var NO_DESCRIPTION = 'no description';
 
-function StateAccessError(message, details) {
+/**
+ * State error class
+ *
+ * @param {*} message
+ * @param {*} details
+ */
+function StateError(message, details) {
```

Comment:
Not a blocker at all, especially since this already existed. But ES6 classes work miracles when extending `Error`.